src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -165,6 +165,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- controls warn as error -->
     <_BinaryFormatterObsoleteAsError>true</_BinaryFormatterObsoleteAsError>
     <!-- controls runtime behavior (AppContext & trimming) -->
+    <EnableUnsafeBinaryFormatterSerialization Condition="'$(EnableUnsafeBinaryFormatterSerialization)' == '' AND '$(_ProjectTypeRequiresBinaryFormatter)' == 'true'">true</EnableUnsafeBinaryFormatterSerialization>
```

Comment:
At this point, EnableUnsafeBinaryFormatterSerialization is known to not be true. It could be empty, false, or something else.  You'll only set it to true if it was empty before not if it was explicitly set to false. Confirming that that's what you want.

Also, this will only run on 7.0+ so confirming you only want this change to apply there

Reply:
Winforms/WPF application might want to disable binary formatter explicitly by setting this property to false, we shouldn't override it.

Reply:
> At this point, EnableUnsafeBinaryFormatterSerialization is known to not be true. It could be empty, false, or something else. You'll only set it to true if it was empty before not if it was explicitly set to false. Confirming that that's what you want.

Ideally I don't want this line to be present _at all_. It's only a workaround for https://github.com/dotnet/sdk/issues/32969. Once that fix is in, I'm reverting this change, since I don't want applications to be carrying the "true" flag in their own runtimeconfig.json files unless they've explicitly set the property themselves as true.

But like Tanya said, for now, we need to support people having explicitly written "true" or "false" in config and we need to honor those explicit settings.

> Also, this will only run on 7.0+ so confirming you only want this change to apply there

Correct. 7.0 WinForms apps are not affected.